lib/e2e/src/error.rs

```diff
@@ -1,13 +1,37 @@
-use alloy::{contract::Error, sol_types::SolError};
+use alloy::sol_types::SolError;
 
 pub trait ErrorExt<E> {
     /// Checks that `Self` corresponds to the typed abi-encoded error
     /// `expected`.
-    fn is(&self, expected: E) -> bool;
+    fn is_err(&self, expected: E) -> bool;
```

Comment:
I named it `ErrorExt::is()`, because of a few reasons:

- It's defined in an `ErrorExt` trait.
- It is implemented for error-like types, like `alloy::contract::Error` and `eyre::ErrReport`.
- At the call site, it looks like this `err.is(/* Some Error */);`

So it's quite clear we are dealing with errors.

And example API that looks like this is the `Object.is()` from javascript. I think this is better than something like `Object.is_object()`. wdyt?

Reply:
Well, thinking a bit more about it, we could make it more similar to the contracts by doing something like this:

```rust
/// An error representing a panic.
pub trait Panic {
    /// Checks that `Self` corresponds to a panic with code `code`.
    fn panicked_with(&self, code: i64) -> bool;
}

/// An error representing a revert with some data.
pub trait Revert<E> {
    /// Checks that `Self` corresponds to the typed abi-encoded error
    /// `expected`.
    fn reverted_with(&self, expected: E) -> bool;
}
```

And then we keep a list of possible panic codes, [like this](https://github.com/NomicFoundation/hardhat/blob/main/packages/hardhat-chai-matchers/src/internal/reverted/panic.ts), which is what the Solidity contracts [use](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/83c7e45092dac350b070c421cd2bf7105616cf1a/test/token/ERC20/ERC20.test.js#L60).

Reply:
Will do!

Reply:
Regarding `is` and `is_err` I agree that `is` is a better option, but I had a name collision for `eyre::ErrReport`. Will try traverse the `ErrReport` to `TransportError` and then I can revert `err.is()`.

Reply:
We made a long a way: `err.assert(...)` -> `err.is(...)` -> `err.is_err(...)` -> ... Curious what will be next:)
`err.is(...)` seems good for me

Reply:
naming is hard 😢